### DISCUSSION TITLE
QASR TTS FastSpeech2 + Conformer +  JETS

### DISCUSSION THREAD

**Discussion by BassamMattar** [espnet/espnet#5021] [Q&A]
Is this normal that training loss begin to converge at this high value? I have tried many different set of hyperparameters but the loss converges at high value each time 
![image](https://user-images.githubusercontent.com/37397559/225746726-71ef549c-c22d-431c-bad9-c19be6676e44.png)
Configuration used:
```# This EXPERIMENTAL configuration is for ESPnet2 to train
# Conformer FastSpeech2 + HiFiGAN vocoder jointly. To run
# this config, you need to specify "--tts_task gan_tts"
# option for tts.sh at least and use 22050 hz audio as the
# training data (mainly tested on LJspeech).
# This configuration tested on 4 GPUs with 12GB GPU memory.
# It takes around 1.5 weeks to finish the training but 100k
# iters model should generate reasonable results.

##########################################################
#                  TTS MODEL SETTING                     #
##########################################################
tts: joint_text2wav
tts_conf:
    # text2mel related
    text2mel_type: fastspeech2
    text2mel_params:
        adim: 384         # attention dimension
        aheads: 2         # number of attention heads
        elayers: 4        # number of encoder layers
        eunits: 1536      # number of encoder ff units
        dlayers: 4        # number of decoder layers
        dunits: 1536      # number of decoder ff units
        positionwise_layer_type: conv1d   # type of position-wise layer
        positionwise_conv_kernel_size: 3  # kernel size of position wise conv layer
        duration_predictor_layers: 2      # number of layers of duration predictor
        duration_predictor_chans: 256     # number of channels of duration predictor
        duration_predictor_kernel_size: 3 # filter size of duration predictor
        postnet_layers: 5                 # number of layers of postnset
        postnet_filts: 5                  # filter size of conv layers in postnet
        postnet_chans: 256                # number of channels of conv layers in postnet
        use_masking: True                 # whether to apply masking for padded part in loss calculation
        encoder_normalize_before: True    # whether to perform layer normalization before the input
        decoder_normalize_before: True    # whether to perform layer normalization before the input
        reduction_factor: 1               # reduction factor
        encoder_type: conformer           # encoder type
        decoder_type: conformer           # decoder type
        conformer_rel_pos_type: latest               # relative positional encoding type
        conformer_pos_enc_layer_type: rel_pos        # conformer positional encoding type
        conformer_self_attn_layer_type: rel_selfattn # conformer self-attention type
        conformer_activation_type: swish             # conformer activation type
        use_macaron_style_in_conformer: true         # whether to use macaron style in conformer
        use_cnn_in_conformer: true                   # whether to use CNN in conformer
        conformer_enc_kernel_size: 7                 # kernel size in CNN module of conformer-based encoder
        conformer_dec_kernel_size: 31                # kernel size in CNN module of conformer-based decoder
        init_type: xavier_uniform                    # initialization type
        transformer_enc_dropout_rate: 0.2            # dropout rate for transformer encoder layer
        transformer_enc_positional_dropout_rate: 0.2 # dropout rate for transformer encoder positional encoding
        transformer_enc_attn_dropout_rate: 0.2       # dropout rate for transformer encoder attention layer
        transformer_dec_dropout_rate: 0.2            # dropout rate for transformer decoder layer
        transformer_dec_positional_dropout_rate: 0.2 # dropout rate for transformer decoder positional encoding
        transformer_dec_attn_dropout_rate: 0.2       # dropout rate for transformer decoder attention layer
        pitch_predictor_layers: 5                    # number of conv layers in pitch predictor
        pitch_predictor_chans: 256                   # number of channels of conv layers in pitch predictor
        pitch_predictor_kernel_size: 5               # kernel size of conv leyers in pitch predictor
        pitch_predictor_dropout: 0.5                 # dropout rate in pitch predictor
        pitch_embed_kernel_size: 1                   # kernel size of conv embedding layer for pitch
        pitch_embed_dropout: 0.0                     # dropout rate after conv embedding layer for pitch
        stop_gradient_from_pitch_predictor: true     # whether to stop the gradient from pitch predictor to encoder
        energy_predictor_layers: 2                   # number of conv layers in energy predictor
        energy_predictor_chans: 256                  # number of channels of conv layers in energy predictor
        energy_predictor_kernel_size: 3              # kernel size of conv leyers in energy predictor
        energy_predictor_dropout: 0.5                # dropout rate in energy predictor
        energy_embed_kernel_size: 1                  # kernel size of conv embedding layer for energy
        energy_embed_dropout: 0.0                    # dropout rate after conv embedding layer for energy
        stop_gradient_from_energy_predictor: false   # whether to stop the gradient from energy predictor to encoder

    # vocoder related
    vocoder_type: hifigan_generator
    vocoder_params:
        out_channels: 1
        channels: 512
        global_channels: -1
        kernel_size: 7
        upsample_scales: [8, 8, 2, 2]
        upsample_kernel_sizes: [16, 16, 4, 4]
        resblock_kernel_sizes: [3, 7, 11]
        resblock_dilations: [[1, 3, 5], [1, 3, 5], [1, 3, 5]]
        use_additional_convs: true
        bias: true
        nonlinear_activation: "LeakyReLU"
        nonlinear_activation_params:
            negative_slope: 0.1
        use_weight_norm: true

    # discriminator related
    discriminator_type: hifigan_multi_scale_multi_period_discriminator
    discriminator_params:
        scales: 1
        scale_downsample_pooling: "AvgPool1d"
        scale_downsample_pooling_params:
            kernel_size: 4
            stride: 2
            padding: 2
        scale_discriminator_params:
            in_channels: 1
            out_channels: 1
            kernel_sizes: [15, 41, 5, 3]
            channels: 128
            max_downsample_channels: 1024
            max_groups: 16
            bias: True
            downsample_scales: [2, 2, 4, 4, 1]
            nonlinear_activation: "LeakyReLU"
            nonlinear_activation_params:
                negative_slope: 0.1
            use_weight_norm: True
            use_spectral_norm: False
        follow_official_norm: False
        periods: [2, 3, 5, 7, 11]
        period_discriminator_params:
            in_channels: 1
            out_channels: 1
            kernel_sizes: [5, 3]
            channels: 32
            downsample_scales: [3, 3, 3, 3, 1]
            max_downsample_channels: 1024
            bias: True
            nonlinear_activation: "LeakyReLU"
            nonlinear_activation_params:
                negative_slope: 0.1
            use_weight_norm: True
            use_spectral_norm: False

    # loss function related
    generator_adv_loss_params:
        average_by_discriminators: false # whether to average loss value by #discriminators
        loss_type: mse                   # loss type, "mse" or "hinge"
    discriminator_adv_loss_params:
        average_by_discriminators: false # whether to average loss value by #discriminators
        loss_type: mse                   # loss type, "mse" or "hinge"
    use_feat_match_loss: true            # whether to use feat match loss
    feat_match_loss_params:
        average_by_discriminators: false # whether to average loss value by #discriminators
        average_by_layers: false         # whether to average loss value by #layers of each discriminator
        include_final_outputs: true      # whether to include final outputs for loss calculation
    use_mel_loss: true     # whether to use mel-spectrogram loss
    mel_loss_params:
        fs: 22050          # must be the same as the training data
        n_fft: 1024        # fft points
        hop_length: 256    # hop size
        win_length: null   # window length
        window: hann       # window type
        n_mels: 80         # number of Mel basis
        fmin: 0            # minimum frequency for Mel basis
        fmax: null         # maximum frequency for Mel basis
        log_base: null     # null represent natural log
    lambda_text2mel: 1.0   # loss scaling coefficient for text2mel loss
    lambda_adv: 1.0        # loss scaling coefficient for adversarial loss
    lambda_mel: 45.0       # loss scaling coefficient for Mel loss
    lambda_feat_match: 2.0 # loss scaling coefficient for feat match loss

    # others
    sampling_rate: 22050          # needed in the inference for saving wav
    segment_size: 32              # segment size for random windowed discriminator
    cache_generator_outputs: true # whether to cache generator outputs in the training

# extra module for additional inputs
pitch_extract: dio           # pitch extractor type
pitch_extract_conf:
    reduction_factor: 1
pitch_normalize: global_mvn  # normalizer for the pitch feature
energy_extract: energy       # energy extractor type
energy_extract_conf:
    reduction_factor: 1
energy_normalize: global_mvn # normalizer for the energy feature
# continue training
#init_param:
#/content/drive/MyDrive/ESPNET_EXP/TTS/JETS/tts_fastspeech2_conformer_JETS_hifigan_noise_reduced/34epoch.pth
##########################################################
#            OPTIMIZER & SCHEDULER SETTING               #
##########################################################
# optimizer setting for generator
optim: adamw
optim_conf:
    lr: 5.0e-5
    betas: [0.9, 0.999]
    eps: 1.0e-9
    weight_decay: 0
scheduler: exponentiallr
scheduler_conf:
    gamma: 0.9
# optimizer setting for discriminator
optim2: adamw
optim2_conf:
    lr: 5.0e-5
    betas: [0.5, 0.999]
    eps: 1.0e-9
    weight_decay: 0
scheduler2: exponentiallr
scheduler2_conf:
    gamma: 0.9
generator_first: true # whether to start updating generator first

##########################################################
#                OTHER TRAINING SETTING                  #
##########################################################
#num_iters_per_epoch: 1000 # number of iterations per epoch
max_epoch: 1000           # number of epochs
accum_grad: 1             # gradient accumulation
batch_bins: 1500000       #batch_size: 8       # batch bins (feats_type=raw)
batch_type: numel         # how to make batch
#batch_size: 1
#batch_type: unsorted         # how to make batch
grad_clip: 2             # gradient clipping norm
grad_noise: false         # whether to use gradient noise injection
sort_in_batch: descending # how to sort data in making batch
sort_batch: descending    # how to sort created batches
num_workers: 4            # number of workers of data loader
use_amp: false            # whether to use pytorch amp
log_interval: 50          # log interval in iterations
keep_nbest_models: 2      # number of models to keep
num_att_plot: 3           # number of attention figures to be saved in every check
seed: 777                 # random seed number
patience: null            # patience for early stopping
unused_parameters: true   # needed for multi gpu case
best_model_criterion:     # criterion to save the best models
-   - valid
    - text2mel_loss
    - min
-   - train
    - text2mel_loss
    - min
-   - train
    - total_count
    - max
cudnn_deterministic: false # setting to false accelerates the training speed but makes it non-deterministic
                           # in the case of GAN-TTS training, we strongly recommend setting to false
cudnn_benchmark: false     # setting to true might acdelerate the training speed but sometimes decrease it
                           # therefore, we set to false as a default (recommend trying both cases)
```
